libdnf/rpm/repo.cpp

```diff
@@ -92,83 +84,105 @@ const int COUNTME_VERSION = 0;
+    LrExceptionWithSourceUrl(int code, const std::string & msg, std::string source_url)
+        : LrException(code, msg)
+        , source_url(std::move(source_url)) {}
+    const std::string & get_source_url() const { return source_url; }
+
 private:
-    std::string sourceUrl;
+    std::string source_url;
 };
 
-static void throwException(std::unique_ptr<GError> && err)
-{
+/// Recursive renames/moves file/directory.
+/// Implements copy and remove fallback.
+static void move_recursive(const std::string & src, const std::string & dest) {
+    try {
+        std::filesystem::rename(src, dest);
+    } catch (const std::filesystem::filesystem_error & ex) {
+        std::filesystem::copy(src, dest, std::filesystem::copy_options::recursive);
+        std::filesystem::remove_all(src);
+    }
+}
+
+static int64_t mtime(const char * filename) {
+    struct stat st;
+    stat(filename, &st);
+    return st.st_mtime;
+}
+
+static void throw_exception(std::unique_ptr<GError> && err) {
     throw LrException(err->code, err->message);
 }
 
-template<typename T>
-inline static void handleSetOpt(LrHandle * handle, LrHandleOption option, T value)
-{
-    GError * errP{nullptr};
-    if (!lr_handle_setopt(handle, &errP, option, value)) {
-        throwException(std::unique_ptr<GError>(errP));
+template <typename T>
+inline static void handle_set_opt(LrHandle * handle, LrHandleOption option, T value) {
+    GError * err_p{nullptr};
+    if (!lr_handle_setopt(handle, &err_p, option, value)) {
+        throw_exception(std::unique_ptr<GError>(err_p));
     }
 }
 
-inline static void handleGetInfo(LrHandle * handle, LrHandleInfoOption option, void * value)
-{
-    GError * errP{nullptr};
-    if (!lr_handle_getinfo(handle, &errP, option, value)) {
-        throwException(std::unique_ptr<GError>(errP));
+inline static void handle_get_info(LrHandle * handle, LrHandleInfoOption option, void * value) {
+    GError * err_p{nullptr};
+    if (!lr_handle_getinfo(handle, &err_p, option, value)) {
+        throw_exception(std::unique_ptr<GError>(err_p));
     }
 }
 
-template<typename T>
-inline static void resultGetInfo(LrResult * result, LrResultInfoOption option, T value)
-{
-    GError * errP{nullptr};
-    if (!lr_result_getinfo(result, &errP, option, value)) {
-        throwException(std::unique_ptr<GError>(errP));
+template <typename T>
+inline static void result_get_info(LrResult * result, LrResultInfoOption option, T value) {
+    GError * err_p{nullptr};
+    if (!lr_result_getinfo(result, &err_p, option, value)) {
+        throw_exception(std::unique_ptr<GError>(err_p));
     }
 }
 
-/* Callback stuff */
+// Callback stuff
 
-int RepoCB::progress(double totalToDownload, double downloaded) { return 0; }
-void RepoCB::fastestMirror(FastestMirrorStage stage, const char * ptr) {}
-int RepoCB::handleMirrorFailure(const char * msg, const char * url, const char * metadata) { return 0; }
+int RepoCB::progress(double /*totalToDownload*/, double /*downloaded*/) {
```

Comment:
I am sorry but I do not understand `/*totalToDownload*/`. It uses `/*` to skip test of clang? If it is like that I don't like it.

Reply:
Unused arguments in parent class (can be used in inherited class) are commented out to suppress compiler warnings (errors in our case). I can remove they completely but the comments say what the arguments means. It is good practise to keep the names in the comments.
C++ 17 defines attribute `[[maybe_unused]]` for that situation, but I can't use it with swig. As I mentioned in previous comment.

Reply:
IMO you can keep the names in headers and leave them out in implementation if unused... but the comment is fine with me too...

Reply:
@lukash I wrote a comment that swig parses only public headers. And there is only one occurence in repo.hpp. And I can move it to implementation file. And I see now, you have the same idea.